server/src/test/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequestTests.java

```diff
@@ -118,65 +112,4 @@ public void testMappingKeyedByType() throws IOException {
             assertEquals(request1.mappings(), request2.mappings());
         }
     }
-
-    @Override
-    protected PutIndexTemplateRequest createTestInstance() {
-        PutIndexTemplateRequest request = new PutIndexTemplateRequest();
-        request.name("test");
-        if (randomBoolean()) {
-            request.version(randomInt());
-        }
-        if (randomBoolean()) {
-            request.order(randomInt());
-        }
-        request.patterns(Arrays.asList(generateRandomStringArray(20, 100, false, false)));
-        int numAlias = between(0, 5);
-        for (int i = 0; i < numAlias; i++) {
-            // some ASCII or Latin-1 control characters, especially newline, can lead to
-            // problems with yaml parsers, that's why we filter them here (see #30911)
-            Alias alias = new Alias(randomRealisticUnicodeOfLengthBetween(1, 10).replaceAll("\\p{Cc}", ""));
-            if (randomBoolean()) {
-                alias.indexRouting(randomRealisticUnicodeOfLengthBetween(1, 10));
-            }
-            if (randomBoolean()) {
-                alias.searchRouting(randomRealisticUnicodeOfLengthBetween(1, 10));
-            }
-            request.alias(alias);
-        }
-        if (randomBoolean()) {
-            try {
-                request.mapping("doc", XContentFactory.jsonBuilder().startObject()
-                    .startObject("doc").startObject("properties")
-                    .startObject("field-" + randomInt()).field("type", randomFrom("keyword", "text")).endObject()
-                    .endObject().endObject().endObject());
-            } catch (IOException ex) {
-                throw new UncheckedIOException(ex);
-            }
-        }
-        if (randomBoolean()) {
-            request.settings(Settings.builder().put("setting1", randomLong()).put("setting2", randomTimeValue()).build());
-        }
-        return request;
-    }
-
-    @Override
-    protected PutIndexTemplateRequest doParseInstance(XContentParser parser) throws IOException {
-        return new PutIndexTemplateRequest().source(parser.map());
```

Comment:
If we don't test parsing for this request any more, we should maybe unit  test the source(Map<String, Object> templateSource) method directly here. There's a bunch of code in there that e.g. throws exceptions sets request properties. I think we call this in the `PutIndexTemplateRequestTests` in the client module, but for good unit testing coverage I think we should add independent testing to this class as well.
This could be done in this PR or in a follow up issue as far as I'm concerned.

Reply:
👍 will add a couple tests around `PutIndexTemplateRequest#source`.